resources/acl.rb

```diff
@@ -8,8 +8,10 @@
   # As we're using the accumulator pattern we need to shove everything
   # into the root run context so each of the sections can find the parent
   with_run_context :root do
+    default_conf
     edit_resource(:template, config_file) do |new_resource|
-      cookbook 'haproxy'
+      source node['haproxy']['conf_template_source']
```

Comment:
I like what you're trying to do here buuuuut...

So we really don't want to be using node attributes here, as it messes with the resources "interface". It should be accessed only one way, and that's via the resource properties. 

The helper library you've added mirrors the `defaults` action of a resource :)